### PR TITLE
DataNode stores segment msgpack positions

### PR DESCRIPTION
**Pull request by congqixia**
DataNode stores segment positions from MsgPack

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/datanode/collection_replica.go [hunk 1]
```diff
@@ -338,3 +348,30 @@ func (replica *CollectionSegmentReplica) hasCollection(collectionID UniqueID) bo
...
+
+// setStartPositions set segment `Start Position` - means the `startPositions` from the MsgPack when segment is first found
+func (replica *CollectionSegmentReplica) setStartPositions(segID UniqueID, startPositions []*internalpb.MsgPosition) error {
+	replica.posMu.Lock()
+	defer replica.posMu.Unlock()
+	replica.startPositions[segID] = startPositions
+	return nil
+}
+
+// setEndPositions set segment `End Position` - means the `startPositions` from the MsgPack when segment need to be flushed
```
Comment (XuanYang-cn) [inline internal/datanode/collection_replica.go:360]: I'm not sure about this. Maybe need @xige-16 's help.

Comment (congqixia) [reply internal/datanode/collection_replica.go]: According to @neza2017, segment End Position shall be the endPositions from MsgPack.

**Review by XuanYang-cn — COMMENTED** [review]
/approve

**Review by sunby — APPROVED** [review]

**Review by neza2017 — COMMENTED** [review]
/approve

**Pull request merged**